### PR TITLE
add dusk test for #250

### PR DESCRIPTION
**Pull request by EvanDime**
##### _List all issues this PR resolves_
- applies to #250

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### CODE REVIEW

File: resources/js/Components/Form/DateTimePicker.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <ctk-date-time-picker v-model="proxyValue" :minute-interval="minuteInterval" :format="format">
+  <ctk-date-time-picker dusk="date-time-picker" :id="id" v-model="proxyValue" :minute-interval="minuteInterval" :format="format">
```
Comment (a-drew) [inline resources/js/Components/Form/DateTimePicker.vue:2]: kind of curious why have an ambiguous dusk name here? if there's more than one on the same page then the dusk tag won't work.

File: resources/js/Pages/Admin/BookingRequests/CreateBookingRequestModal.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
     <jet-dialog-modal :show="room" @close="closeModal" max-width="">
         <template #content>
 
-          <div class="overflow-y-auto h-96">
+          <div :dusk="'createBookingModal'" class="overflow-y-auto h-96">
```
Comment (a-drew) [inline resources/js/Pages/Admin/BookingRequests/CreateBookingRequestModal.vue:5]: you can remove the binding and just use the string directly

File: tests/Browser/Components/DateTimePicker.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if ($newMonth){
+            $browser->click('#'.$this->id.'-picker-container-DatePicker > div > div.datepicker-controls.flex.align-center.justify-content-center > div.arrow-month.h-100.text-right > button')->pause(1000);
+        }
+        $browser->click('#'.$this->id.'-picker-container-DatePicker > div > div.month-container > span > div > button:nth-child('.$day.')');
+        $browser
+            ->click('> div.datetimepicker.flex.visible > div > div.pickers-container.flex > div.time-picker.flex.flex-fixed.flex-1.with-border > div.time-picker-column.flex-1.flex.flex-direction-column.text-center.time-picker-column-hours > div > button:nth-child('.$hour.')')
+            ->click('> div.datetimepicker.flex.visible > div > div.pickers-container.flex > div.time-picker.flex.flex-fixed.flex-1.with-border > div.time-picker-column.flex-1.flex.flex-direction-column.text-center.time-picker-column-minutes > div > button:nth-child(2)')
+            ->click('> div.datetimepicker.flex.visible > div > div.datepicker-buttons-container.flex.justify-content-right.button-validate.flex-fixed > button.datepicker-button.validate.flex.align-center.justify-content-center')
```
Comment (a-drew) [inline tests/Browser/Components/DateTimePicker.php:67]: oh boi

**Review by alecadub — APPROVED** [review]
THANKS ! dateinput will be a joke now.

**Review by Alex-Hayeur — APPROVED** [review]
Good work, the dusk test was obviously the priority but the other small refactors are super helpful.

**Comment by EvanDime** [comment]
> THANKS ! dateinput will be a joke now.

will maybe add thing to put a specific time instead of 00 for minutes

**Review by a-drew — APPROVED** [review]
amazing work figuring out how we can feed into into the datetime picker.

File: resources/js/Components/Form/DateTimePicker.vue (hunk 1 above)
Comment (EvanDime) [reply resources/js/Components/Form/DateTimePicker.vue:2]: If was going to be used for the dusk component but we ended up using the id. If someone only has 1 on a page and doesn't want to put an id they can use the dusk tag.

File: resources/js/Pages/Admin/BookingRequests/CreateBookingRequestModal.vue (hunk 2 above)
Comment (EvanDime) [reply resources/js/Pages/Admin/BookingRequests/CreateBookingRequestModal.vue:5]: wasn't sure at first if i would need to generate an id number next to it at first.

File: tests/Browser/Components/DateTimePicker.php (hunk 3 above)
Comment (EvanDime) [reply tests/Browser/Components/DateTimePicker.php:67]: i've seen some shit...

**Pull request merged**